### PR TITLE
Vendorengrams redux

### PR DESCRIPTION
**Pull request by 48klocs**
This addresses issue #4849 - the return of vendorengrams.xyz integration was done with olde fashioned services, this updates things to use Redux.

### CODE REVIEW

File: src/app/shell/Header.tsx [hunk 1]
```diff
@@ -338,6 +339,9 @@ class Header extends React.PureComponent<Props, State> {
     }
 
     this.updateVendorEngrams(this.props.account || undefined);
+
+    const vendorEngramDropActive = this.props.vendorDrops.some(isDroppingHigh);
```
Comment (bhollis) [inline src/app/shell/Header.tsx]: Put this in your mapStateToProps function. That way, you'll only re-render when `vendorEngramDropActive` changes, not when any of vendorDrops changes. Plus, your component will depend on the minimum state.

File: src/app/shell/Header.tsx [hunk 2]
```diff
@@ -338,6 +339,9 @@ class Header extends React.PureComponent<Props, State> {
...
+    this.setState({ vendorEngramDropActive });
```
Comment (bhollis) [inline src/app/shell/Header.tsx]: You don't need to set this in state - it's already available as a prop.

File: src/app/vendorEngramsXyzApi/reducer.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        vendorDrops: [],
+        lastUpdated: unloadedDate
+      };
+    }
+    default:
+      return state;
+  }
+};
+
+export function saveVendorDropsToIndexedDB() {
```
Comment (bhollis) [inline src/app/vendorEngramsXyzApi/reducer.ts]: You probably want to also save the last updated time, so you can ignore any saved data that's too old.

File: src/app/vendorEngramsXyzApi/reducer.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  );
+}
+
+export function loadVendorDropsFromIndexedDB(): ThunkResult<Promise<void>> {
+  return async (dispatch, getState) => {
+    if (!getState().vendorDrops.loaded) {
+      const vendorDrops = await get<VendorDropsState>('vendorengrams');
+
+      // easing the transition from the old state (just an array) to the new state
+      // (object containing an array)
```
Comment (bhollis) [inline src/app/vendorEngramsXyzApi/reducer.ts]: I don't get this comment

File: src/app/vendors/Vendor.tsx [hunk 5]
```diff
@@ -43,7 +38,7 @@ export default function Vendor({
   ).join(', ');
 
   const vendorEngramDrops = $featureFlags.vendorEngrams
-    ? getVendorDropsForVendor(vendor.def.hash, allVendorEngramDrops)
+    ? store.getState().vendorDrops.vendorDrops.filter((vd) => vd.vendorId === vendor.def.hash)
```
Comment (bhollis) [inline src/app/vendors/Vendor.tsx]: You should not directly access the store like this - it won't cause updates to the component when the store changes. Either pass this info in as a prop, or connect this component and fish it out of the store using `mapStateToProps`.

File: src/app/vendors/Vendors.tsx [hunk 6]
```diff
@@ -323,8 +317,7 @@ function VendorGroup({
   ownedItemHashes,
   currencyLookups,
   defs,
-  filtering,
-  vendorEngramDrops
```
Comment (bhollis) [inline src/app/vendors/Vendors.tsx]: This is still present in the props definition for this component, and you probably still want it to pass into `Vendor`.

File: src/app/vendorEngramsXyzApi/reducer.ts (hunk 4 above)
Comment (48klocs) [reply src/app/vendorEngramsXyzApi/reducer.ts]: Yeah, I think that was a comment from the old form (maybe?).

File: src/app/vendorEngramsXyzApi/reducer.ts (hunk 3 above)
Comment (48klocs) [reply src/app/vendorEngramsXyzApi/reducer.ts]: This uses the `dropsNeedRefresh` method now to see if the state we've pulled is outdated or not.

File: src/app/vendors/Vendor.tsx (hunk 5 above)
Comment (48klocs) [reply src/app/vendors/Vendor.tsx]: Passing it in now.

**Pull request merged**